src/dawn/SIR/SIR.h

```diff
@@ -314,9 +314,14 @@ struct Value : NonCopyable {
 
```

Comment:
Is the default constructor needed (and related `TypeKind::None`)? I am wondering how this would be used, because it complicates the implementation of Value and any user of Value (because they all need to scope with the possibility that Value might be nothing), e.g. CodeGen.cpp, Line 67.

Reply:
The default constructor is not needed and neither is `TypeKind::None`. What is needed is the `empyt()` method for things like `bool foo`, i.e. variables with a type but no value. However, in this case I believe that `empty()` is a bad name for the method at hand and I opted to rename it to `has_value()`

Reply:
What's the reason for an empty Value? Respectively, how can a type have an empty set of possible values?